MISRA.md

```diff
@@ -0,0 +1,37 @@
+| Rule 3.1 | Required | Allow nested comments. C++ style `//` comments are used in example code within Doxygen documentation blocks. |
+| Rule 11.5 | Advisory | Allow casts from `void *`. Fields such as publish payloads are passed as `void *` and must be cast to the correct data type before use. |
+| Rule 21.1 | Required | Allow use of all macro names. For compatibility, some macros introduced in C99 are defined for use with C90 compilers. |
+| Rule 21.2 | Required | Allow use of all macro and identifier names. For compatibility, some macros introduced in C99 are defined for use with C90 compilers. |
+
+### Flagged by Coverity
+| Deviation | Category | Justification |
+| :-: | :-: | :-: |
+| Rule 5.7 | Required | Type for network context is defined by each transport implementation; however only one transport implementation is needed by an application at a time. If the transport implementations are analyzed one at a time by Coverity, this violation will not be flagged. |
+| Rule 8.7 | Advisory | API functions are not used by the library outside of the files they are defined; however, they must be externally visible in order to be used by an application. |
+
+### Suppressed with Coverity Comments
+| Deviation | Category | Justification |
+| :-: | :-: | :-: |
+| Directive 4.6 | Advisory | Basic numerical types are used in platform implementations, since POSIX platform functions and openssl functions used in the platform implementations take arguments of basic numerical types. |
+| Rule 10.1 | Required | POSIX specific macro utility is flagged for this violation. This macro utility, whose implementation is supplied by the system, is used in the transport implementation. |
+| Rule 10.8 | Required | POSIX specific macro utility is flagged for this violation. This macro utility, whose implementation is supplied by the system, is used in the transport implementation. |
+| Rule 11.3 | Required | Transport implementation casts a pointer of a object type to a pointer of a different object type. This casting is supported in POSIX and is used to obtain the IP address from a address record. |
+| Rule 11.8 | Required | An openssl API used in TLS transport implementation internally casts the pointer to a string literal to a `void *` pointer. |
+| Rule 13.4 | Required | POSIX specific macro utility is flagged for this violation. This macro utility, whose implementation is supplied by the system, is used in the transport implementation. |
+| Rule 14.4 | Required | POSIX specific macro utility is flagged for this violation. This macro utility, whose implementation is supplied by the system, is used in the transport implementation. |
+| Rule 21.6 | Required | The Standard Library input/output functions for opening and closing a file are used by the openssl transport implementation, as the openssl API to read a PEM file takes `FILE *` as an argument. |
```

Comment:
```suggestion
| Rule 21.6 | Required | The Standard Library input/output functions for opening and closing files are used by the openssl transport implementation, since the openssl API to read PEM files takes `FILE *` as an argument. |
```